src/hooks/useMarketsData.ts

```diff
@@ -166,8 +178,51 @@ export const useMarketsData = (
         );
       });
 
+    if (unlaunchedMarkets.data && !hideUnlaunchedMarkets && testFlags.pml) {
+      const unlaunchedMarketsData = Object.values(unlaunchedMarkets.data).map((market) => {
+        const { id, name, logo, sectorTags, price, percentChange24h, tickSizeDecimals } = market;
+
+        if (listOfMarkets.some((m) => m.assetId === id)) return null;
+
+        return safeAssign(
+          {},
+          {
+            id: `${id}-USD`,
+            assetId: id,
+            displayId: `${getDisplayableAssetFromBaseAsset(id)}-USD`,
+            clobPairId: -1,
+            effectiveInitialMarginFraction: LIQUIDITY_TIERS[4].initialMarginFraction,
+            imageUrl: logo,
+            initialMarginFraction: LIQUIDITY_TIERS[4].initialMarginFraction,
+            isNew: false,
+            isUnlaunched: true,
+            line: undefined,
+            name,
+            nextFundingRate: undefined,
+            openInterest: undefined,
+            openInterestUSDC: undefined,
+            oraclePrice: price,
+            priceChange24H: price && percentChange24h ? price * percentChange24h : undefined,
+            priceChange24HPercent: percentChange24h,
+            tags: sectorTags ?? [],
+            tickSizeDecimals,
+            trades24H: 0,
+            volume24H: 0,
+          }
+        );
+      });
+
+      return [...listOfMarkets, ...unlaunchedMarketsData.filter(isTruthy)];
+    }
+
     return listOfMarkets;
-  }, [allPerpetualClobIds, allPerpetualMarkets, allAssets, sevenDaysSparklineData]);
+  }, [
+    allPerpetualClobIds,
+    allPerpetualMarkets,
+    allAssets,
+    sevenDaysSparklineData,
+    unlaunchedMarkets,
```

Comment:
slightly prefer unlaunchedMarkets.data so we're not churning every time the loading state changes